src/components/QuizCard.js

```diff
@@ -145,71 +148,90 @@ const QuizCard = ({ onAnswer, onFinish, ...props }) => {
     }
   };
 
-  console.log("currentTresh", currentTresh, store.tresholds);
-
   return (
-    <div className={`${styles.container} quiz-card`}>
-      {store.getAnswer === true && store.isInstantFeedback ? <Answer /> : null}
+    <>
 
       {store.showFinalScore === false && questions.length > 0 ? (
         <>
-          <h1 className={styles.quiz_title_card}>
-            {questions[currentQuestion].title}
-            {props.debug && (
-              <div className={styles.debugScore}>
-                Pos: {questions[currentQuestion].position}
+          {store.getAnswer === true && store.isInstantFeedback ?
+            <Answer />
+            :
+            <>
+              <div className={styles.quiz_card_top_wrapper}>
+
+                <div>
+                  <p className={styles.progress}>{currentQuestion}/{questions.length}</p>
+                </div>
+
+                <h1 className={styles.quiz_card_title}>
+                  {questions[currentQuestion].title}
+                  {props.debug && (
+                    <div className={styles.debugScore}>
+                      Pos: {questions[currentQuestion].position}
+                    </div>
+                  )}
+                </h1>
+
+                <div>
+                  <p className={styles.timer}>
+                    <IconBase viewBox='0 0 24 15'>
+                      {timerIconPath}
+                    </IconBase>
+                    {store.timer} sec
+                  </p>
+                </div>
               </div>
-            )}
-          </h1>
 
-          <div className={styles.quiz_grid}>
-            {Array.isArray(questions[currentQuestion].options) &&
-              questions[currentQuestion].options.map((option, i) => {
-                return (
-                  <Fragment key={i}>
-                    {questions[currentQuestion].question_type === "SELECT" ? (
-                      <button
-                        key={option.id}
-                        name="isSelect"
-                        onClick={() => selectAnswer(option)}
-                        className={styles.quiz_card_option}>
-                        <h2 className={styles.buttonTextSelector}>
-                          {option.title}
-                        </h2>
-                        {props.debug && (
-                          <div className={styles.debugScore}>
-                            <p className="m-0 p-0">Score: {option.score}</p>
-                            <p className="m-0 p-0">Pos: {option.position}</p>
-                          </div>
+              <div className={styles.quiz_grid}>
+                {Array.isArray(questions[currentQuestion].options) &&
+                  questions[currentQuestion].options.map((option, i) => {
+                    return (
+                      <Fragment key={i}>
+                        {questions[currentQuestion].question_type === "SELECT" ? (
```

Comment:
You shouldn't add nested ternary conditions, it makes the code harder to read. It's better another component that manages each of this cases with if statements, then you just call that component in the code